napari/utils/_injection.py

```diff
@@ -1,30 +1,35 @@
 from functools import wraps
 from inspect import isgeneratorfunction, signature
-from typing import TYPE_CHECKING, Any, Callable, Dict, Optional, Type, TypeVar
+from typing import Any, Callable, Dict, Optional, Type, TypeVar
 
 from typing_extensions import get_type_hints
 
-if TYPE_CHECKING:
-    from ..layers import Layer
+from .. import components, layers, viewer
+from ..viewer import current_viewer
 
 T = TypeVar("T")
+_NULL = object()
 
 
-def _get_active_layer() -> Optional['Layer']:
-    from ..viewer import current_viewer
+def _get_active_layer() -> Optional[layers.Layer]:
+    return v.layers.selection.active if (v := current_viewer()) else None
 
-    if viewer := current_viewer():
-        return viewer.layers.selection.active
 
+def _get_active_layer_list() -> Optional[components.LayerList]:
+    return v.layers if (v := current_viewer()) else None
 
-def _get_active_layer_list() -> Optional['Layer']:
-    from ..viewer import current_viewer
 
-    if viewer := current_viewer():
-        return viewer.layers
+# registry of Type -> "accessor function"
+# where each value is a function that is capable
+# of retrieving an instance of it's corresponding key type.
+_ACCESSORS: Dict[Type, Callable[..., Optional[object]]] = {
+    layers.Layer: _get_active_layer,
+    viewer.Viewer: current_viewer,
+    components.LayerList: _get_active_layer_list,
+}
```

Comment:
to be sure I'm following, this is our internal mapping of type -> accessor function?

Reply:
yep